src/main/java/org/mifos/connector/mpesa/flowcomponents/transaction/CollectionResponseProcessor.java

```diff
@@ -35,8 +35,11 @@ public void process(Exchange exchange) {
         Object hasTransferFailed = exchange.getProperty(TRANSACTION_FAILED);
 
         if (hasTransferFailed != null && (boolean)hasTransferFailed) {
+            String body = exchange.getIn().getBody(String.class);
             variables.put(TRANSACTION_FAILED, true);
-            variables.put(ERROR_INFORMATION, exchange.getIn().getBody(String.class));
+            variables.put(ERROR_INFORMATION, body);
+            exchange.setProperty(TRANSACTION_FAILED, true);
+            exchange.setProperty(ERROR_INFORMATION, body);
         } else {
             variables.put(TRANSACTION_FAILED, false);
 
```

Comment:
@SubhamPramanik, LINE 56... variables are not set in zeebe.

Reply:
Why do you have 2 of message publishers here? Shouldn't it be only one with the appropriate name. 


zeebeClient.newPublishMessageCommand()
                    .messageName(TRANSFER_MESSAGE)
                    .correlationKey(exchange.getProperty(TRANSACTION_ID, String.class))
                    .variables(variables)
                    .send()
                    .join();
        }

        logger.info("Publishing transaction message variables: " + variables);

        zeebeClient.newPublishMessageCommand()
                .messageName(TRANSFER_RESPONSE)
                .correlationKey(exchange.getProperty(TRANSACTION_ID, String.class))
                .timeToLive(Duration.ofMillis(timeToLive))
                .variables(variables)
                .send()
                .join();

Reply:
okay, so the issue is with the messageName, since in the workflow I have set the TRANSFER_MESSAGE only the above one is working, thanks for the lead.. I will check this and inform you if its working or not

Reply:
<img width="1440" alt="Screenshot 2021-12-01 at 2 20 45 PM" src="https://user-images.githubusercontent.com/31315800/144202081-6b6dff7a-3631-4383-b77d-e5b59344ba85.png">

also the message name will be `transaction-request` as in the SS rit?

Reply:
Yes

Reply:
I am still getting the same issue

Reply:
## In below SS the publishMessageIs executed successfully, 
<img width="1440" alt="Screenshot 2021-12-01 at 4 21 36 PM" src="https://user-images.githubusercontent.com/31315800/144221531-d0906f3d-911c-43c9-a106-5a41b447889a.png">

## But take a look in below SS the variable is not set and flow stuck at gateway.

<img width="1440" alt="Screenshot 2021-12-01 at 4 22 25 PM" src="https://user-images.githubusercontent.com/31315800/144221631-dd6cf07f-6e65-4889-ae91-0886f84ba586.png">